trillian/integration/logenv.go

```diff
@@ -98,7 +98,11 @@ func NewCTLogEnv(ctx context.Context, cfgs []*configpb.LogConfig, numSequencers
 			}
 		}
 		http.Handle("/metrics", promhttp.Handler())
-		server.Serve(listener)
+		if err := server.Serve(listener); err != nil {
+			// An error is expected when the server shuts down. If it fails to
+			// actually serve then tests will fail.
```

Comment:
Alternatively, could check whether `err != http.ErrServerClosed`  and then log the error.

Reply:
From looking at the log it was a network error from accept rather than `ErrServerClosed`. Possibly it's not been shut down cleanly.